pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -111,20 +154,171 @@ func keyPairPEM() (string, string, error) {
 	return pubKeyPEM.String(), privKeyPEM.String(), nil
 }
 
-func localJWTForKey(id string, privateKkeyPem string) (string, error) {
+// generateToken returns a JWT token based on the private key.
+func generateToken(id string, privateKkeyPem string, expiry ...time.Time) (string, error) {
+	// Create the JWT claims
+	claims := jwt.MapClaims{
+		string(InfraEnvKey): id,
+	}
+
+	// Set the expiry time if provided
+	if len(expiry) > 0 {
+		claims["exp"] = expiry[0].Unix()
+	}
+
+	// Create the token using the ES256 signing method and the claims
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, claims)
+
 	priv, err := jwt.ParseECPrivateKeyFromPEM([]byte(privateKkeyPem))
 	if err != nil {
 		return "", err
 	}
+	// Sign the token with the provided private key
+	tokenString, err := token.SignedString(priv)
+	if err != nil {
+		return "", err
+	}
+	return tokenString, nil
+}
+
+func initClient(kubeconfig string) (*kubernetes.Clientset, error) {
+	var err error
+	var config *rest.Config
+	if kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, err
+	}
 
-	token := jwt.NewWithClaims(jwt.SigningMethodES256, jwt.MapClaims{
-		string(InfraEnvKey): id,
-	})
+	k8sclientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
 
-	tokenString, err := token.SignedString(priv)
+	return k8sclientset, err
+}
+
+func (a *AuthConfig) createOrUpdateAuthTokenSecret(kubeconfigPath string) error {
+	k8sclientset, err := initClient(kubeconfigPath)
+	if err != nil {
+		return err
+	}
+	// check if secret exists
+	retrievedSecret, err := k8sclientset.CoreV1().Secrets(authTokenSecretNamespace).Get(context.Background(), authTokenSecretName, metav1.GetOptions{})
+	// if the secret does not exist
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return a.createSecret(k8sclientset)
+		}
+		// Other errors while trying to get the secret
+		return fmt.Errorf("unable to retrieve secret %s/%s: %w", authTokenSecretNamespace, authTokenSecretName, err)
+	}
+
+	// if the secret exists in the cluster, get the token
+	retrievedToken, err := extractAuthTokenFromSecret(retrievedSecret)
+	if err != nil {
+		return err
+	}
+	expiryTime, err := ParseExpirationFromToken(retrievedToken)
+	if err != nil {
+		return err
+	}
+	// Calculate 24 hours before the expiration time
+	thresholdTime := expiryTime.Add(-24 * time.Hour)
+	// Check if current time is past the thresholdTime time of 24 hours
+	if time.Now().After(thresholdTime) {
```

Comment:
The token was created as [UTC](https://github.com/openshift/installer/pull/8717/files#diff-174c7f0a2bed95144baa2b38b752cb3a48aab7bc4380c16028faed7dd97924f9R91). Given that the `oc adm node-image monitor` command can be executed on a different machine (with different date time) than the one used for the `oc adm node-image create` one, I guess it's better to stick on UTC comparison:

```suggestion
	if time.Now().UTC().After(thresholdTime) {
```